activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,17 @@
+*   `AR::Relation#reverse_order` throws `AR::IrreversibleOrderError` when the order can not be reversed
+    using current trivial algorithm. Also raises the same error when `#reverse_order` is called on
+    relation without any order and table has no primary key:
+
+        Topic.order("concat(author_name, title)").reverse_order
+          # Before: SELECT `topics`.* FROM `topics` ORDER BY concat(author_name DESC, title) DESC
+          # After: raises AR::IrreversibleOrderError
+        Edge.all.reverse_order
+          # Before: SELECT `edges`.* FROM `edges` ORDER BY `edges`.`` DESC
+          # After: raises AR::IrreversibleOrderError
```

Comment:
Here, and in the example just above, you have to spell out `ActiveRecord` as a matter of good documentation.